### PR TITLE
chore: update gas fee controller to 15.1.2

### PR DESCRIPTION
**Pull request by smilingkylan**
## **Description**

This PR upgrades `gas-fee-controller` from version 13 to version 15.1.2 and fixes the breaking changes associated with the upgrade.

## **Related issues**

Fixes: https://github.com/MetaMask/mobile-planning/issues/1785

## **Manual testing steps**

1. Create a transaction to send tokens from one wallet to another. Verify that the gas fee looks agreeable.

### Regular Transactions
Before:
Screenshot 2024-05-20 at 6 04 52 PM

After:
Screenshot 2024-05-20 at 5 54 26 PM

### Dapp Transaction
Before: 
Screenshot 2024-05-21 at 5 59 42 PM

After:
Screenshot 2024-05-21 at 6 12 47 PM

Reported prices (app vs Etherscan)
ETH Mainnet: .00017 ETH vs 0.00017017
Linea mainnet: < .00001 ETH vs .000001 ETH
Sepolia testnet: .00023 ETH vs 0.00022519

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/util/test/initial-background-state.json [hunk 1]
```diff
@@ -204,7 +204,8 @@
     "gasFeeEstimates": {},
     "estimatedGasFeeTimeBounds": {},
     "gasEstimateType": "none",
-    "gasFeeEstimatesByChainId": {}
+    "gasFeeEstimatesByChainId": {},
+    "nonRPCGasFeeApisDisabled": false
```
Comment (Cal-L) [inline app/util/test/initial-background-state.json:208]: Does this need a migration?

**Review by Cal-L — COMMENTED** [review]
What was the conclusion for the failing E2E for 1559 gas estimate type? Was it resolved on the API side?

**Comment by kylanhurt** [comment]
Checked migration and it appears to be working just fine 👍

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**